### ISSUE TITLE
curl 127.0.0.1:8053/dns-query returns json response  status 2 comment invalid argument value and doh server doesn't seem to work

### ISSUE THREAD

**Issue by sanfx** [m13253/dns-over-https#90]
Hello ,

I appreciate you have put lot of effort in put this up together, I have been trying to setup DoH on my home network from more than 2 weeks now. going through your GitHub page I see some hope.

I pulled your repository followed steps `make` and then `sudo make install ` on ubuntu running on arm64  architecture 
now when I try curl 127.0.0.1:8053/dns-query I get response saying:
`{"Status":2,"Comment":"Invalid argument value: \"ct\" = \"\""}`

I have setup this with apache2, I already had domain name self hosted on ssl so I had cert I can use it .

so even when I tried https://selfhosted.ddns.com/dns-query I got the same json result.\
I think the problem is on the doh server end to fix this json response.

I used this self hosted.ddns.com/dns-query in chrome://settings > advanced > Security > custom DNS  and tried from google chrome by visiting 1.1.1.1/help but I still do not get DoH .

Any help will be greatly appreciated thank you

**Comment by satishweb** [comment]
I think you need to provide query parameters to get dns resolution. You might want to try the docker based tutorial for easy setup. Readme document has all steps recorded.

**Comment by satishweb** [comment]
This is the example of curl call:
`curl -w '\n' 'https://dns.example.com/dns-query?name=google.com&type=A'`

**Comment by sanfx** [comment]
ah, this works, so does that mean I can use this as upstream resolver ? 
i.e. selfhosted.ddns.com/dns-query  in the doh-client configuration ?

**Comment by satishweb** [comment]
I am not sure if I understood your question correctly. However, the DoH server will always need at least one upstream DNS server address to forward queries. It could be a custom DNS server address such as 8.8.8.8 (Google) or your ISP DNS server address or your own hosted DNS server on your system.

It appears to me that you are trying to get the DoH server working for your home network and use the same DoH while you are outside of your home network (DDNS + Router port NAT).

If I have a correct understanding of your requirements then you might find this tutorial helpful: https://github.com/satishweb/docker-doh/blob/master/README.md

Note: The tutorial makes use of Lets Encrypt SSL certs (with DNS based validation on R53). If you want to use your certs, you will have to remove the cert-manager service and configure proxy service to use your certs directly.

**Comment by sanfx** [comment]
Thanks Satish, yes you are right however I already have a SSL certs of a self hosted domain name 

NOTE: https://selfhosted.ddns.com is not original pasted here

curl -w '\n' 'https://selfhosted.ddns.com/dns-query'  results 

{"Status":0,"TC":false,"RD":true,"RA":true,"AD":false,"CD":false,"Question":[{"name":"google.com.","type":1}],"Answer":[{"name":"google.com.","type":1,"TTL":207,"Expires":"Wed, 30 Sep 2020 20:09:57 UTC","data":"216.58.210.206"}]}

but when I entered https://selfhosted.ddns.com/dns-query in the google chrome  


![doh-server](https://user-images.githubusercontent.com/2231415/94735426-a67fbe00-0362-11eb-95d4-a51b4ec8cc08.png)

**Comment by sanfx** [comment]
No worries I got it, I had doh-client resolving DNS over https from cloudflare dns , so I used udp:ip_of_doh_client:5300 in the doh-server.conf upstream list as first item and rest as fallback

**Comment by satishweb** [comment]
Not sure why it's not working for you in google chrome (I am assuming that you do not have any cert errors when your DoH url is accessed in browser). Your curl command output shows that it is resolving the DNS name correctly and that means DoH server setup is fine.

In my case, I wanted all of my applications to use DoH so I installed cloudflared DoH client and I did not go through the hassle of configuring each browser/application with DoH settings.
